include/deal.II/base/vectorization.h

```diff
@@ -180,6 +180,11 @@ class VectorizedArray
    */
   static const unsigned int n_array_elements = 1;
 
+  static_assert(
+    width == n_array_elements,
+    "There seems to be something wrong with "
+    "The base variant of this class only supports a single entry in the array, i.e., width=1!");
```

Comment:
I will extend the doxygen class documentation during the day.

To your questions. `VectorizedArray<Number, width>` is specilized for 8 cases:
```cpp
VectorizedArray<double, 1>
VectorizedArray<double, 2>
VectorizedArray<double, 4>
VectorizedArray<double, 8>
VectorizedArray<float, 1>
VectorizedArray<float, 4>
VectorizedArray<float, 8>
VectorizedArray<float, 16>
```
If you specify any other `width`, the compiler will go to the default implementation (which has `n_array_elements=1`) and will through a static error since `width == n_array_elements` is not fulfilled.

Reply:
Ok. Then we should update the error message above to be more helpful (as in: you specified an illegal width that is not supported).